### PR TITLE
added loans heap comments

### PR DESCRIPTION
**Pull request by ith-harvey**

### CODE REVIEW

File: src/libraries/internal/Loans.sol [hunk 1]
```diff
@@ -33,12 +33,20 @@ library Loans {
...
     function init(LoansState storage loans_) internal {
         loans_.loans.push(Loan(address(0), 0));
     }
 
+    /**
+    * The Loans heap is a Max Heap data structure (complete binary tree), the root node is the loan with the highest threshold price (TP)
+    * at a given time. The heap is represented as an array, where the first element is a dummy element (Loan(address(0), 0)) and the first
+    * value of the heap starts at index 1, ROOT_INDEX. The threshold price of a loan's parent is always greater than or equal to the
+    * threshold price of the loan.
+    * This code was modified from the following source: https://github.com/zmitton/eth-heap/blob/master/contracts/Heap.sol
```
Comment (EdNoepel) [inline src/libraries/internal/Loans.sol:47]: Should we change this file to MIT license and cite in _Licensing_ section of README?

**Review by EdNoepel — APPROVED** [review]

File: src/libraries/internal/Loans.sol (hunk 1 above)
Comment (ith-harvey) [reply src/libraries/internal/Loans.sol:47]: Happy to do this... so at the top of the file I would do:
```
// SPDX-License-Identifier: MIT
```

Then when changes do you suggest in the licensing section of the readme?

Comment (ith-harvey) [reply src/libraries/internal/Loans.sol:47]: Or you can feel free to add to this branch, whatever is easiest : )

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**